### PR TITLE
[Libunwind] Try to fix msan failures

### PR DESCRIPTION
**Pull request by chestnykh**

### CODE REVIEW

**Comment by chestnykh** [comment]
@ldionne i run these tests locally with msan (x86-64) and have no errors. Could you please tell on what architecture(s) the tests was failing?

**Comment by chestnykh** [comment]
All CI failures look unrelated

**Comment by ldionne** [comment]
Rebased onto `main` to re-trigger CI, which should solve the errors you were seeing.

I don't remember why these `XFAIL`s were added, but if the CI passes with them removed, I'm happy.

**Review by ldionne — APPROVED** [review]
LGTM if the CI is green.

**Comment by chestnykh** [comment]
Maybe load order breaks something:
With `-lunwind` passed explicitly:
```
        linux-vdso.so.1 (0x0000700799647000)
        libunwind.so.1 => /usr/lib/libunwind.so.1 (0x00007007995fe000)
        libatomic.so.1 => /usr/lib/libatomic.so.1 (0x00007007995f3000)
        libc++.so.1 => /usr/lib/libc++.so.1 (0x00007007994ec000)
        libc++abi.so.1 => /usr/lib/libc++abi.so.1 (0x00007007949c0000)
        libm.so.6 => /usr/lib/libm.so.6 (0x00007007948d1000)
        libresolv.so.2 => /usr/lib/libresolv.so.2 (0x00007007948bf000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007007946ce000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007007946a0000)
        /lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x0000700799649000)
```

With only `-rtlib=compiler-rt -unwindlib=libunwind` (without explicit `-lunwind` in the driver cmdline)
```
        linux-vdso.so.1 (0x000077ed1ce4a000)
        libunwind.so.1 => /usr/lib/libunwind.so.1 (0x000077ed1ce01000)
        libatomic.so.1 => /usr/lib/libatomic.so.1 (0x000077ed1cdf6000)
        libc++.so.1 => /usr/lib/libc++.so.1 (0x000077ed1ccef000)
        libc++abi.so.1 => /usr/lib/libc++abi.so.1 (0x000077ed181c0000)
        libm.so.6 => /usr/lib/libm.so.6 (0x000077ed180d1000)
        libresolv.so.2 => /usr/lib/libresolv.so.2 (0x000077ed180bf000)
        libc.so.6 => /usr/lib/libc.so.6 (0x000077ed17ece000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x000077ed17ea0000)
        /lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x000077ed1ce4c000)
```

**Review by arichardson — CHANGES_REQUESTED** [review]
We shouldn't need to initialize these structures to silence sanitizer errors. I had a draft pr last year to fix this properly but forgot about it since other things came up: https://github.com/llvm/llvm-project/pull/67860

I'd prefer something like that. Alternatively I can update my change once I'm back in the office in January.

File: libunwind/src/UnwindLevel1-gcc-ext.c [hunk 1]
```diff
@@ -133,6 +133,10 @@ _LIBUNWIND_EXPORT _Unwind_Reason_Code
 _Unwind_Backtrace(_Unwind_Trace_Fn callback, void *ref) {
   unw_cursor_t cursor;
   unw_context_t uc;
+#if __has_feature(memory_sanitizer)
+  __builtin_memset(&cursor, 0, sizeof(cursor));
```
[the remark this answers is not in the document]
Comment (chestnykh) [reply libunwind/src/UnwindLevel1-gcc-ext.c:137]: It's better to create a bug on msan?

**Pull request closed without merging**